File: 3-arrays/1-intro/main.cpp
// Arrays - C++ Example

#include <iostream>

int main() {
    int arr[] = {1, 2, 3, 4, 5};

    std::cout << "Element at index 2: " << arr[2] << std::endl;  // Output: 3

    arr = 10;
    std::cout << "Modified array: ";
    for (int i = 0; i < 5; i++) {
        std::cout << arr[i] << ' ';
    }
    std::cout << std::endl;

    return 0;
}

File: 3-arrays/1-intro/main.java
// Arrays - Java Example

class ArraysIntro {

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};

        System.out.println("Element at index 2: " + arr[2]); // Output: 3

        // arr = 10; // This line would cause a compile-time error: Type mismatch: cannot convert from int to int[]

        System.out.print("Modified array: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
